plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -929,7 +929,7 @@ public void createTable(Table table)
         catch (AlreadyExistsException e) {
             throw new TableAlreadyExistsException(new SchemaTableName(table.getDbName(), table.getTableName()));
         }
-        catch (NoSuchObjectException e) {
+        catch (NoSuchObjectException | InvalidObjectException e) {
```

Comment:
Why are we doing this change?
In case it is relevant, it should be done in a separate commit.

Reply:
I have removed this change and then I get the following error:

```
Error:    TestDeltaLakeOssDeltaLakeConnectorTest>BaseConnectorTest.testCreateViewSchemaNotFound:898->AbstractTestQueryFramework.assertQueryFails:398 
Expecting message:
  <"test_schema_1k9beg2gw3">
to match regex:
  <"Schema test_schema_1k9beg2gw3 not found">
but did not.

Throwable that failed the check:

io.trino.testing.QueryFailedException: test_schema_1k9beg2gw3
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:122)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:481)
	at io.trino.testing.QueryAssertions.assertQueryFails(QueryAssertions.java:290)
	at io.trino.testing.AbstractTestQueryFramework.assertQueryFails(AbstractTestQueryFramework.java:398)
	at io.trino.testing.BaseConnectorTest.testCreateViewSchemaNotFound(BaseConnectorTest.java:898)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
	Suppressed: java.lang.Exception: SQL: CREATE VIEW test_schema_1k9beg2gw3.test_view_create_no_schema_1bfq4rcibk AS SELECT 1 AS c1
		at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:484)
		... 16 more
	Suppressed: java.lang.Exception: Query: CREATE VIEW test_schema_1k9beg2gw3.test_view_create_no_schema_1bfq4rcibk AS SELECT 1 AS c1
		at io.trino.testing.QueryAssertions.assertQueryFails(QueryAssertions.java:294)
		... 15 more
Caused by: io.trino.spi.TrinoException: test_schema_1k9beg2gw3
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.createTable(ThriftHiveMetastore.java:936)
	at io.trino.plugin.hive.metastore.thrift.BridgingHiveMetastore.createTable(BridgingHiveMetastore.java:201)
	at io.trino.plugin.hive.metastore.cache.CachingHiveMetastore.createTable(CachingHiveMetastore.java:501)
	at io.trino.plugin.hive.ViewsMetastoreHelper.createView(ViewsMetastoreHelper.java:103)
	at io.trino.plugin.deltalake.DeltaLakeMetadata.createView(DeltaLakeMetadata.java:2063)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata.createView(ClassLoaderSafeConnectorMetadata.java:564)
	at io.trino.metadata.MetadataManager.createView(MetadataManager.java:1243)
	at io.trino.execution.CreateViewTask.execute(CreateViewTask.java:117)
	at io.trino.execution.CreateViewTask.execute(CreateViewTask.java:46)
	at io.trino.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:145)
	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:249)
	at io.trino.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:143)
	at io.trino.$gen.Trino_testversion____20221030_172723_5726.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.apache.hadoop.hive.metastore.api.InvalidObjectException: test_schema_1k9beg2gw3
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_table_result$create_table_resultStandardScheme.read(ThriftHiveMetastore.java:54409)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_table_result$create_table_resultStandardScheme.read(ThriftHiveMetastore.java:54386)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_table_result.read(ThriftHiveMetastore.java:54312)
	at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:86)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.recv_create_table(ThriftHiveMetastore.java:1539)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.create_table(ThriftHiveMetastore.java:1526)
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastoreClient.createTable(ThriftHiveMetastoreClient.java:237)
	at io.trino.plugin.hive.metastore.thrift.FailureAwareThriftMetastoreClient.lambda$createTable$7(FailureAwareThriftMetastoreClient.java:132)
	at io.trino.plugin.hive.metastore.thrift.FailureAwareThriftMetastoreClient.runWithHandle(FailureAwareThriftMetastoreClient.java:467)
	at io.trino.plugin.hive.metastore.thrift.FailureAwareThriftMetastoreClient.createTable(FailureAwareThriftMetastoreClient.java:132)
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.lambda$createTable$36(ThriftHiveMetastore.java:924)
	at io.trino.plugin.hive.metastore.thrift.ThriftMetastoreApiStats.lambda$wrap$0(ThriftMetastoreApiStats.java:42)
	at io.trino.plugin.hive.util.RetryDriver.run(RetryDriver.java:130)
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.createTable(ThriftHiveMetastore.java:922)
	... 15 more
```

Reply:
See related comment https://github.com/trinodb/trino/pull/14769#discussion_r1006269172

cc @ebyhr

Reply:
Adressed in the meantime through #14769